lighthouse-cli/test/smokehouse/smokehouse-report.js

```diff
@@ -98,6 +98,18 @@ function findDifference(path, actual, expected) {
     }
   }
 
+  // If the expected value is an array, assert on the length.
+  // This still allows for asserting that the first n elements of an array are specified elements,
+  // but requires using an object literal (ex: {0: x, 1: y, 2: z} matches [x, y, z, q, w, e] and
+  // {0: x, 1: y, 2: z, length: 5} does not match [x, y, z].
+  if (Array.isArray(expected) && actual.length !== expected.length) {
+    return {
+      path,
```

Comment:
should we do 
```js
path: `${path}.length`,
```

?

agreed that printing the full array value is best though